docs/devguide/building/index.rst

```diff
@@ -104,6 +104,68 @@ If configure runs successfully it will end with a message about the next steps f
 
     setenv
 
+.. _building--configuration-header:
+
+The Configuration Header
+========================
+
+The :ghfile:`configure` script and CMake generate a header ``$DDS_ROOT/OpenDDSConfig.h`` (see :envvar:`DDS_ROOT`).
+The header defines macros that determine the features and capabilities that are built into OpenDDS.
+Users using a custom build environment must ensure that ``$DDS_ROOT/OpenDDSConfig.h`` exists.
+A template can be found in :ghfile:`dds/OpenDDSConfig.h.in`.
+
+The configuration header is not used directly.
+Instead, :ghfile:`dds/OpenDDSConfigWrapper.h` includes it, provides defaults and backwards compatibility, and performs some sanity checks.
+For consistency and to avoid build errors, applications that have code that is conditioned on OpenDDS features should include :ghfile:`dds/OpenDDSConfigWrapper.h`.
+
+The following macros are available in the config header:
+
+``OPENDDS_CONFIG_AUTO_STATIC_INCLUDES``
+    TODO
```

Comment:
TODO

Reply:
Yeah.  So what is a decent blurb for this?

Reply:
This is the current explanation in a section that talks about the libraries:

https://github.com/OpenDDS/OpenDDS/blob/62c78ee53f7556b99c517d192fe4fa2d99f394d5/docs/devguide/introduction.rst?plain=1#L458

Reply:
Yes, it's what's doing the "auto including" if `OPENDDS_CONFIG_AUTO_STATIC_INCLUDES` is true:

https://github.com/OpenDDS/OpenDDS/blob/62c78ee53f7556b99c517d192fe4fa2d99f394d5/dds/DCPS/StaticIncludes.h#L6-L12